miniupnpd/minissdp.c

```diff
@@ -65,33 +65,28 @@ AddMulticastMembership(int s, struct lan_addr_s * lan_addr)
 	struct ip_mreqn imr;	/* Ip multicast membership */
 #endif	/* HAVE_IP_MREQN */
 
-    /* setting up imr structure */
-    imr.imr_multiaddr.s_addr = inet_addr(SSDP_MCAST_ADDR);
-    /*imr.imr_interface.s_addr = htonl(INADDR_ANY);*/
+	/* setting up imr structure */
+	imr.imr_multiaddr.s_addr = inet_addr(SSDP_MCAST_ADDR);
 #ifndef HAVE_IP_MREQN
 	imr.imr_interface.s_addr = lan_addr->addr.s_addr;
 #else	/* HAVE_IP_MREQN */
-    imr.imr_address.s_addr = lan_addr->addr.s_addr;
+	imr.imr_address.s_addr = lan_addr->addr.s_addr;
 #ifndef MULTIPLE_EXTERNAL_IP
 #ifdef ENABLE_IPV6
 	imr.imr_ifindex = lan_addr->index;
 #else	/* ENABLE_IPV6 */
-    imr.imr_ifindex = if_nametoindex(lan_addr->ifname);
+	imr.imr_ifindex = if_nametoindex(lan_addr->ifname);
 #endif	/* ENABLE_IPV6 */
 #else	/* MULTIPLE_EXTERNAL_IP */
-    imr.imr_ifindex = 0;
+	imr.imr_ifindex = 0;
 #endif	/* MULTIPLE_EXTERNAL_IP */
 #endif	/* HAVE_IP_MREQN */
 
-#ifndef HAVE_IP_MREQN
-	if (setsockopt(s, IPPROTO_IP, IP_ADD_MEMBERSHIP, (void *)&imr, sizeof(struct ip_mreq)) < 0)
-#else	/* HAVE_IP_MREQN */
-	if (setsockopt(s, IPPROTO_IP, IP_ADD_MEMBERSHIP, (void *)&imr, sizeof(struct ip_mreqn)) < 0)
-#endif	/* HAVE_IP_MREQN */
+	if (setsockopt(s, IPPROTO_IP, IP_ADD_MEMBERSHIP, (void *)&imr, sizeof(imr)) < 0)
```

Comment:
@miniupnp are you ok with this simplification in the same patch? reason - same approach (just one sizeof(ver) w/o #ifdefs) is used for IP_MULTICAST_IF arg for notify socket.

Reply:
yes it's fine ;)
thank you for the simplification